local-cloudbuild.py

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/env python
+#
+# Run the steps of cloudbuild.yaml locally, emulating Google Container
+# Builder functionality.  Does not actually push images to the
+# Container Registry.  Not all functionality is supported.
+#
+# Based on https://cloud.google.com/container-builder/docs/api/build-steps
+
+# System packages
+import argparse
+import getpass
+import os
+import shutil
+import subprocess
+import sys
+import tempfile
+
+# Third party packages
+import yaml
+
+
+# Types
+class CloudBuildError(Exception):
+    pass
+
+
+def main(argv):
+    """Main entrypoint for cli"""
+    parser = argparse.ArgumentParser(
+        description='Process cloudbuild.yaml locally to build Docker images')
+    parser.add_argument(
+        'cloudbuild', type=str, help='Path to cloudbuild.yaml input file')
+    parser.add_argument(
+        '--keep_workspace',
+        type=bool,
+        default=False,
+        help='Retain workspace directory after building')
+    args = parser.parse_args(argv[1:])
```

Comment:
"You added dependency injection for testability, but didn't actually write any tests." :)

I will add a tests, and move this into a /script subdirectory (that previously held other scripts until I moved them into runtime-image/).